cmake/find/Findodb.cmake

```diff
@@ -0,0 +1,108 @@
+# ----------------------------------------------------------------------
```

Comment:
As far as I can see we already have `odb` package in Hunter so this change can be made separately. Can you please open pull request only with this module and simple example (just `hunter_add_package(odb)` + `find_package(odb REQUIRED)`). Also I see no problem with caching this package - what error do you have exactly? Maybe this package is not relocatable by design, i.e. it hardcoding some path while installing? RPATH?